### PR TITLE
Help Center: clean-up part 1 useSupportAvailability

### PR DESCRIPTION
**Pull request by arcangelini**
## Proposed Changes

Changes the support availability endpoint to support status endpoint. This new endpoint has proper naming to be aligned with our new omni-channel support. The variables and function naming is also updated.

## Why are these changes being made?

Most of this is variable renaming. The larger change is the update endpoint from: `help/eligibility/chat/mine` to: `help/support-status`.

#### Needs to be used with D151949-code

1. Sandbox your public-api
2. Open up Calypso for this branch
3. Check out Help Center to ensure it is working properly
4. Open up a support request to check that messaging is working properly

Part 2 - #91896
Part 3 - #91917

### CODE REVIEW

File: packages/help-center/src/hooks/use-chat-status.ts [hunk 1]
```diff
@@ -14,11 +14,12 @@ export default function useChatStatus(
 	group: MessagingGroup = 'wpcom_messaging',
 	checkAgentAvailability = true
 ) {
-	const { data: chatStatus } = useSupportAvailability();
+	const { data: supportStatus } = useSupportStatus();
+	const { availability } = supportStatus || {};
```
Comment (alshakero) [inline packages/help-center/src/hooks/use-chat-status.ts]:
```suggestion
	const availability = supportStatus.availability;
```

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-status.php
Comment (alshakero) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-status.php:1]: This file is unimported anywhere.

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php
Comment (alshakero) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php:1]: This file is still imported in `class-help-center.php`

**Review by alshakero — CHANGES_REQUESTED** [review]
Works great in Calypso and simple but couldn't test in Atomic because the new file is not imported in `class-help-center.php`

**Review by agrullon95 — APPROVED** [review]
LGTM. Tested on simple and atomic. I was able to open the support chart with the "Contact WordPress.com Support" and send a test message.

**Pull request merged**